XPCBackupService/Entities/BackupTreeNode.swift

```diff
@@ -89,48 +95,89 @@ class BackupTreeNode {
         return attribute[FileAttributeKey.modificationDate] as? Date
     }
 
-    private func nodeIsSynced(url: String, deviceId: Int) throws -> Bool {
-        let realm = try BackupRealm.shared.getRealm()
-        let syncedNode = realm.objects(SyncedNode.self).first { syncedNode in
-            url == syncedNode.url && deviceId == syncedNode.deviceId
+    private func nodeIsSynced(url: URL, deviceId: Int) throws -> ThreadSafeReference<SyncedNode>? {
+        let syncedNode = autoreleasepool {
```

Comment:
This was causing a huge memory usage because we were not deallocating all the memory used to store all the syncedNodes in the Realm query, autoreleasepool fixes the issue

Reply:
great work!